controllers/rcj_soccer_referee_supervisor/referee/supervisor.py

```diff
@@ -376,10 +436,95 @@ def reset_team_for_kickoff(self, team: str):
 
         return robot
+
+    def is_neutral_spot_occupied(self, ns_x: float, ns_z: float) -> bool:
+        """Check whether the specific neutral spot is occupied
+
+        Args:
+            ns_x (float): x position of the neutral spot
+            ns_z (float): z position of the neutral spot
+
+        Returns:
+            bool: Whether the neutral spot is unoccupied
+        """
+        # Check whether any of the robots is blocking the neutral spot
+        for _, pos in self.robot_translation.items():
+            rx, rz = pos[0], pos[2]
+            distance = math.sqrt((rx - ns_x)**2 + (rz - ns_z)**2)
+            if distance < DISTANCE_AROUND_UNOCCUPIED_NEUTRAL_SPOT:
+                return True
+
+        # Check whether the ball is blocking the neutral spot
+        bx, bz = self.ball_translation[0], self.ball_translation[2]
+        distance = math.sqrt((bx - ns_x) ** 2 + (bz - ns_z) ** 2)
+        if distance < DISTANCE_AROUND_UNOCCUPIED_NEUTRAL_SPOT:
+            return True
+
+        return False
+
+    def get_unoccupied_neutral_spot(
+        self,
+        distance_type: NeutralSpotDistanceType,
+        object_name: str,
+    ) -> Optional[str]:
+        """Get the nearest/furthest unoccupied neutral spot.
+
+        Args:
+            distance_type (NeutralSpotDistanceType): Either nearest or furthest
+            object_pos (Tuple[int, int]): Get the spot for object at this pos
+            robot_name (str, optional
+
+        Returns:
+            str (optional): The name of the best spot. If the value is None
+                there is no free neutral spot.
+        """
+        if object_name == "ball":
+            x = self.ball_translation[0]
+            z = self.ball_translation[2]
+        else:
+            x = self.robot_translation[object_name][0]
+            z = self.robot_translation[object_name][2]
+
+        find_nearest = distance_type == NeutralSpotDistanceType.NEAREST.value
+        best_distance = -1
+        best_spot = None
+        for ns, ns_pos in NEUTRAL_SPOTS.items():
+            ns_x, ns_z = ns_pos
+            spot_distance = math.sqrt((x - ns_x)**2 + (z - ns_z)**2)
+
+            if find_nearest:
+                is_better = best_spot is None or best_distance > spot_distance
+            else:
+                is_better = best_spot is None or best_distance < spot_distance
+
+            if is_better and not self.is_neutral_spot_occupied(ns_x, ns_z):
+                best_distance = spot_distance
+                best_spot = ns
+
+        return best_spot
+
+    def move_object_to_neutral_spot(
+        self,
+        object_name: str,
+        neutral_spot: Optional[str],
+    ):
+        """Move the robot to the specified neutral spot.
+
+        Args:
+            object_name (str): Name of the object (Ball or robot's name)
+            neutral_spot (str, optional): The spot the robot will be moved to
+        """
+        if neutral_spot is None:
```

Comment:
@Adman I don't see this function being used without the neutral spot provided, nor does it seem like it would actually do anything, if `neutral_spot` was not provided.

Do you have some future plans with this function?

Reply:
@mrshu There is one edge case. Imagine all the robots and ball are located on neutral spots, so you cannot move the object anywhere.